src/helper/helper.js

```diff
@@ -416,6 +407,10 @@ helper.merge_ = function(from, to) {
       } else if (plain.isPlainObject(fromProperty)) {
         if (!plain.isPlainObject(to[property])) to[property] = {};
         helper.merge_(fromProperty, to[property]);
+      } else if (plain.hasOwn(to, property)) {
+        if (fromProperty !== undefined) {
+          to[property] = fromProperty;
+        }
```

Comment:
I don't like these lines since they break backwards compatibility. Rather it may be better to update the documentation to be more clear. It's confusing what it means by 'undefined' values since with an array there are kind of 2 types of undefined: values in an array that have not been defined, and values in an array that have been defined as undefined. I actually ran into this issue and added a test for it: See lines 245 to 256 in merge_test.js for what I mean (actually, changing this may break that test).